cmake/pika_setup_cuda.cmake

```diff
@@ -143,19 +143,15 @@ if(PIKA_WITH_CUDA AND NOT TARGET pika_internal::cuda)
 endif()
 
 function(pika_add_nvhpc_cuda_flags source)
-  set_source_files_properties(${source} PROPERTIES LANGUAGE CXX)
-
-  get_source_file_property(source_compile_flags ${source} COMPILE_FLAGS)
-  if(source_compile_flags STREQUAL "NOTFOUND")
-    set(source_compile_flags)
-  endif()
-
   set(source_gpu_cc_flags)
   foreach(cuda_arch ${CMAKE_CUDA_ARCHITECTURES})
     set(source_gpu_cc_flags "${source_gpu_cc_flags} -gpu=cc${cuda_arch}")
   endforeach()
 
+  get_source_file_property(source_compile_flags ${source} COMPILE_FLAGS)
+  string(REPLACE "NOTFOUND" "" source_compile_flags ${source_compile_flags})
```

Comment:
Highly unlikely, but in principle the flags could have something like `-DBLAH_NOTFOUND_123` and the `NOTFOUND` would get removed. Perhaps do a regex with `"^NOTFOUND$"` or simply keep the explicit `if` for this?